media/client/ipc/source/MediaPipelineIpc.cpp

```diff
@@ -226,53 +226,90 @@ bool MediaPipelineIpc::attachSource(const std::unique_ptr<IMediaPipeline::MediaS
     if (configType == SourceConfigType::VIDEO_DOLBY_VISION || configType == SourceConfigType::VIDEO ||
         configType == SourceConfigType::AUDIO)
     {
-        IMediaPipeline::MediaSourceAV &mediaSourceAV = dynamic_cast<IMediaPipeline::MediaSourceAV &>(*source);
-        request.set_segment_alignment(convertSegmentAlignment(mediaSourceAV.getSegmentAlignment()));
+        IMediaPipeline::MediaSourceAV *mediaSourceAV = nullptr;
+        try
+        {
+            mediaSourceAV = &dynamic_cast<IMediaPipeline::MediaSourceAV &>(*source);
```

Comment:
if You use ptrs, it's better to dynamic_cast to ptr. It doesn't throw any exception (dynamic cast to ptr returns nullptr when failed)